src/Deposit.js

```diff
@@ -1099,4 +1103,174 @@ export default class Deposit {
+   * Only callable if the deposit is sufficiently collateralised.
+   */
+  async exitCourtesyCall() {
+    await EthereumHelpers.sendSafely(this.contract.methods.exitCourtesyCall())
+  }
+
+  /**
+   * Notify the contract that the deposit is severely undercollateralised,
+   * and begin liquidation of the signer bonds.
+   *
+   * The bonds are auctioned in a falling-price auction. The value of
+   * the bonds can be queried using `Deposit.auctionValue`, and bids placed
+   * using `Deposit.purchaseSignerBondsAtAuction`.
+   */
+  async notifyUndercollateralizedLiquidation() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifyUndercollateralizedLiquidation()
+    )
+  }
+
+  /**
+   * Purchases the signer bonds and closes the liquidation auction.
+   *
+   * Note: The caller must approve the Deposit contract to spend
+   * `lotSize` amount of TBTC.
+   */
+  async purchaseSignerBondsAtAuction() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.purchaseSignerBondsAtAuction()
+    )
+  }
+
+  /**
+   * Gets the current value of signer bonds at auction.
+   * Only callable if the deposit is in the liqudation state.
+   * @return {BN} auction value in wei.
+   */
+  async auctionValue() {
+    return toBN(await this.contract.methods.auctionValue().call())
+  }
+
+  // /--------------------- Timeout helpers -------------------------
+
+  /**
+   * Notify the contract that signing group setup has timed out.
+   * Only applicable during funding.
+   */
+  async notifySignerSetupFailure() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifySignerSetupFailure()
+    )
+  }
+
+  /**
+   * Notify the contract that the funder has failed to send BTC.
+   * Only applicable during funding.
+   */
+  async notifyFundingTimeout() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifyFundingTimeout()
+    )
+  }
+
+  /**
+   * Notifies the contract that the courtesy period has elapsed.
+   */
+  async notifyCourtesyTimeout() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifyCourtesyTimeout()
+    )
+  }
+
+  /**
+   * Notify the contract that the signers have failed to produce a signature.
+   * This is considered fraud, and moves the deposit into liquidation.
+   * Only applicable during redemption.
+   */
+  async notifySignatureTimeout() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifySignatureTimeout()
+    )
+  }
+
+  /**
+   * Notify the contract that the signers have failed to produce a redemption proof.
+   */
+  async notifyRedemptionProofTimeout() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifyRedemptionProofTimeout()
+    )
+  }
+
+  /**
+   * Checks if signature was requested via the Keep.
+   * @param {string} digest Digest to check approval for.
+   * @return {boolean} True if signature approved, false if not (fraud).
+   */
+  async wasSignatureApproved(digest) {
```

Comment:
I think we can cheat this one by calling `isAwaitingSignature(digest)` on the backing keep.

Reply:
Though I suppose this will false-negative if this signature was approved, but previously (e.g., in case of fee bump). So 50-50 on that.

Reply:
Yeah I don't believe the method is giving us what we'd want - the aim is to use wasSignatureApproved to check fraud. Checking if it's approved would be strictly different.